dt-login/login-endpoints.php

```diff
@@ -81,25 +81,29 @@ public function login( WP_REST_Request $request ) {
         }
         //phpcs:enable
 
+        $error_response = null;
         $user_manager = new DT_Login_User_Manager( $payload );
 
         try {
             $response = $user_manager->login();
         } catch ( \Throwable $th ) {
-            return new WP_Error( $th->getCode(), $th->getMessage(), [ 'status' => 401 ] );
+            $error_response = new WP_Error( $th->getCode(), $th->getMessage(), [ 'status' => 401 ] );
         }
 
         if ( is_wp_error( $response ) ) {
-            return $response;
+            $error_response = $response;
         }
         if ( !$response ) {
-            return new WP_Error( 'login_error', 'Something went wrong with the login', [ 'status' => 401 ] );
+            $error_response = new WP_Error( 'login_error', 'Something went wrong with the login', [ 'status' => 401 ] );
         }
 
-        return new WP_REST_Response( [
+        // Provide opportunity to update response packet.
+        $rest_response = apply_filters( 'dt_sso_login_response', [
             'status' => 200,
             'body' => $response,
-        ] );
+        ], $payload, $error_response );
+
+        return !is_null( $error_response ) ? $error_response : new WP_REST_Response( $rest_response );
```

Comment:
@kodinkat will return the $error_response if it is not null no matter if $rest_response has been updated via filter.
Is this the desired behavior?

Reply:
Yep, so as to maintain the refactored logic-flow; $error_response to take priority.... Previously, it was returned immediately when encountered.

Reply:
I guess, it could be refactored further to ensure if error detected; then no need to apply filter; but, my thinking was to also communicate to others, that an error has occurred..

Reply:
@kodinkat would they not need control over the error response in order apply to their business logic?

Reply:
I might not fully understand the aim. My intuition says just limit it to success responses. Or allow both success and errors to be modified

Reply:
@corsacca yeah, maybe a more generic shape(s), might be to separate out across two new filters (`dt_sso_login_response_success` / `dt_sso_login_response_error`) - Both of which would return their respective success/error array/obj...

Interested parties can then hook in and process accordingly.... what do you think?